torch/_lobpcg.py

```diff
@@ -164,11 +497,43 @@ def lobpcg(A,                   # type: Tensor
 
     """
 
+    # While it is expected that `A` is symmetric,
+    # the `A_grad` might be not. Therefore we perform the trick below,
+    # so that `A_grad` becomes symmetric.
+    # The symmetrization is important for first-order optimization methods,
+    # so that (A - alpha * A_grad) is still a symmetric matrix.
+    # Same holds for `B`.
+    A_sym = (A + A.transpose(-2, -1)) / 2
```

Comment:
I'm not sure about this.
Is A expected to be symmetric already here?
Is this just done just for the backward or benefits the forward as well?
If it's only for the backward, we should move it to the backward function to make sure there is no perf downside for people that don't use it.

Reply:
A is expected to be symmetric, but the gradient is not going to be for sure. The out the test for `symeig_backward` where the same trick is employed. It is only to guarantee that `A.grad` is symmetric.

Reply:
My question is: why isn't this code in the `backward()`? After you compute the gradient, you make it symmetric and return it.
There is no need to do these extra computations at forward time right?

Reply:
Well, but then I have to move it into the test the same way `symeig` does it, otherwise the gradient will be incorrect because you cannot force the symmetry in the backward. The current implementation always guarantees symmetric gradients in such a way that `gradcheck` will never complain. So, what is the best  way?

Reply:
Gradient is never going to be symmetric because the input matrix is not being treated as two equal halfs.

Reply:
Ho I see what you mean.
I guess the perf drop is not significant compared to the rest of the forward computation here?
Also maybe this should be documented? And marked as BC-breaking?

Reply:
I have added a warning. Since I use your code for the forward that handles TorchScript, it is not bc-breaking anymore, I presume.

Reply:
I think you need to change the condition to make this not BC-breaking: only use the custom Function if either A or B requires grad.